src/appleseed/foundation/utility/string.h

```diff
@@ -379,12 +382,21 @@ namespace impl
 }
 
 // General entry-point.
-template <typename T>
+template <typename T, typename std::enable_if_t<!IsPointer<T>::R>* = 0>
```

Comment:
What's the purpose of the default template param value here?

Also, if it turns out to be necessary/useful, could `nullptr` be used instead of `0` (here and below)?

Reply:
This specification is a mistake ...
Can't use that to distinguish pointer and non-pointer.

From the reference: http://en.cppreference.com/w/cpp/types/enable_if

> ### Notes
>
> A common mistake is to declare two function templates that differ only in their default template arguments. This is illegal because default template arguments are not part of function template's signature, and declaring two different function templates with the same signature is illegal.
>
```cpp
struct T {
    enum { int_t,float_t } m_type;
    template <typename Integer,
              typename = std::enable_if_t<std::is_integral<Integer>::value>
    >
    T(Integer) : m_type(int_t) {}
 
    template <typename Floating,
              typename = std::enable_if_t<std::is_floating_point<Floating>::value>
    >
    T(Floating) : m_type(float_t) {} // error: cannot overload
};
```

Reply:
I deleted this specialization because you did it already in your implementation structure, which, I modified in order to return a `"<null>"` string when the `value == nullptr`.